internal/cmd/cmd.go

```diff
@@ -294,3 +306,50 @@ Additional help topics:{{range .Commands}}{{if .IsAdditionalHelpTopicCommand}}
 Use "{{.CommandPath}} [command] --help" for more information about a command.{{end}}
 `
 }
+
+// warnVersion checks if the client version is ahead of the server and logs a warning if so
+func warnVersion(ctx context.Context, api *api.Client) {
+	// Don't bother printing the server version for SaaS
+	if strings.HasSuffix(api.URL, ".infrahq.com") {
+		return // the hosted infra instance is always the latest instance
+	}
+
+	server, err := api.GetServerVersion(ctx)
+	if err != nil {
+		logging.L.Debug().Err(err).Msg("check server version")
+		return
+	}
+
+	ahead, err := versionAhead(api.Version, server.Version)
+	if err != nil {
+		logging.L.Debug().Err(err).Msg("failed to compare client version to server")
+		return
+	}
+	if ahead {
+		fmt.Println("Warning: Your client version is ahead of the server you are connected to.")
+		fmt.Printf("Client: %s\n", api.Version)
+		fmt.Printf("Server: %s\n", server.Version)
+		fmt.Println()
+	}
+}
+
+// versionAhead returns if the current semantic version is comparison version
+func versionAhead(current, compare string) (bool, error) {
+	if current == compare {
```

Comment:
Without this check equivalent versions return true, I believe its due to appending the `-0`.

cc: @jmorganca do we need that `-0` check? If not, this check can go.

Reply:
It was important for "pre-release" versions, but I don't think it's applicable here.